sotodlib/toast/ops/sim_sso.py

```diff
@@ -278,62 +292,96 @@ def _get_sso_position(self, data, sso, times, observer):
-            data,
-            obs,
-            sso_name,
-            sso_az,
-            sso_el,
-            sso_dist,
-            sso_diameter,
-            prefix,
-            dets,
+        self,
+        data,
+        obs,
+        sso_name,
+        sso_az,
+        sso_el,
+        sso_dist,
+        sso_diameter,
+        prefix,
+        dets,
     ):
         """
         Observe the SSO with each detector in tod
         """
         log = Logger.get()
         timer = Timer()
 
-        for det in dets:
+        # Get a view of the data which contains just this single
+        # observation
+        obs_data = data.select(obs_name=obs.name)
+
+        zaxis = np.array([0.0, 0.0, 1.0])
+        bore_quat = obs_data.obs[0].shared[defaults.boresight_azel][:]
+        # print(f"Obs dir = {qa.rotate(bore_quat, zaxis)}")
+        bore_lon, bore_lat, _ = qa.to_lonlat_angles(bore_quat)
+        # print(
+        #     f"Obs lon / lat = {bore_lon} ({bore_lon*180/np.pi}) / {bore_lat} ({bore_lat*180/np.pi})"
+        # )
+        # print(
+        #     f"Obs Az / El = {-bore_lon} ({-bore_lon*180/np.pi}) / {bore_lat} ({bore_lat*180/np.pi})"
+        # )
+
+        for idet, det in enumerate(dets):
             timer.clear()
             timer.start()
             bandpass = obs.telescope.focalplane.bandpass
             signal = obs.detdata[self.det_data][det]
 
-            # Compute detector quaternions
-
-            obs_data = Data(comm=data.comm)
-            obs_data._internal = data._internal
-            obs_data.obs = [obs]
             self.detector_pointing.apply(obs_data, detectors=[det])
-            obs_data.obs.clear()
-            del obs_data
-
-            azel_quat = obs.detdata[self.detector_pointing.quats][det]
+            det_quat = obs_data.obs[0].detdata[self.detector_pointing.quats][det]
 
             # Convert Az/El quaternion of the detector into angles
-            theta, phi = qa.to_position(azel_quat)
+            theta, phi, _ = qa.to_iso_angles(det_quat)
 
             # Azimuth is measured in the opposite direction
             # than longitude
-            az = 2 * np.pi - phi
+            az = -phi
             el = np.pi / 2 - theta
 
+            mean_az = np.mean(az)
+            mean_el = np.mean(el)
+
             # Convolve the planet SED with the detector bandpass
             planet_freq, planet_temp = self._get_planet_temp(sso_name)
             det_temp = bandpass.convolve(det, planet_freq, planet_temp)
 
             beam, radius = self._get_beam_map(det, sso_diameter, det_temp)
 
             # Interpolate the beam map at appropriate locations
+
             x = (az - sso_az.to_value(u.rad)) * np.cos(el)
             y = el - sso_el.to_value(u.rad)
-            r = np.sqrt(x ** 2 + y ** 2)
+            r = np.sqrt(x**2 + y**2)
+
             good = r < radius
             sig = beam(x[good], y[good], grid=False)
             signal[good] += sig
 
+            # if idet == 0:
```

Comment:
Did you intend to drop these debug statements?

Reply:
I removed the commented out debugging statements from the operator itself, but left the commented out plotting function at the bottom of the source.  We could delete that later or move to some other source if it is useful.